src/features/events/store.ts

```diff
@@ -142,6 +142,34 @@ const eventsSlice = createSlice({
         remoteItem(participant.id, { data: participant })
       );
     },
+    participantUpdated: (
+      state,
+      action: PayloadAction<[number, ZetkinEventParticipant]>
+    ) => {
+      const [eventId, participant] = action.payload;
+      const item = state.participantsByEventId[eventId].items.find(
+        (item) => item.id === participant.id
+      );
+
+      if (item) {
+        item.data = { ...item.data, ...participant };
+        item.mutating = [];
+      } else {
+        state.participantsByEventId[eventId].items.push(
+          remoteItem(participant.id, { data: participant })
+        );
+      }
+
+      if (participant.cancelled) {
+        // If cancelled participant was contact for event, also remove contact
+        const event = state.eventList.items.find(
+          (e) => e?.data?.id === eventId
+        );
+        if (event?.data && event?.data?.contact?.id == participant.id) {
```

Comment:
It should be enough to just do this:

```suggestion
        if (event?.data?.contact?.id == participant.id) {
```

But no need to change it now. I'm proposing a new lint rule that would have caught this, and when that's introduced there will be a lot of little fixes like this that can all be handled in one go.